Tests/MongoSwiftSyncTests/SpecTestRunner/TestOperationResult.swift

```diff
@@ -120,3 +149,141 @@ extension DeleteResult: BulkWriteResultConvertible {
+        self.errorLabelsContain = try container.decodeIfPresent([String].self, forKey: .errorLabelsContain)
+        self.errorLabelsOmit = try container.decodeIfPresent([String].self, forKey: .errorLabelsOmit)
+    }
+
+    internal static func == (lhs: ErrorResult, rhs: ErrorResult) -> Bool {
+        lhs.errorContains == rhs.errorContains &&
+            lhs.errorCodeName == rhs.errorCodeName &&
+            lhs.errorLabelsContain == rhs.errorLabelsContain &&
+            lhs.errorLabelsOmit == rhs.errorLabelsOmit
+    }
+
+    public func checkErrorResult(_ error: Error) throws {
+        try self.checkErrorContains(error)
+        try self.checkCodeName(error)
+        try self.checkErrorLabels(error)
+    }
+
+    // swiftlint:disable cyclomatic_complexity
+
+    internal func checkErrorContains(_ error: Error) throws {
+        if let errorContains = self.errorContains?.lowercased() {
+            if let commandError = error as? CommandError {
+                expect(commandError.message.lowercased()).to(contain(errorContains))
+            } else if let writeError = error as? WriteError {
+                if let writeFailure = writeError.writeFailure {
+                    expect(writeFailure.message.lowercased()).to(contain(errorContains))
+                }
+                if let writeConcernFailure = writeError.writeConcernFailure {
+                    expect(writeConcernFailure.message.lowercased()).to(contain(errorContains))
+                }
+            } else if let bulkWriteError = error as? BulkWriteError {
+                if let writeFailures = bulkWriteError.writeFailures {
+                    for writeFailure in writeFailures {
+                        expect(writeFailure.message.lowercased()).to(contain(errorContains))
+                    }
+                }
+                if let writeConcernFailure = bulkWriteError.writeConcernFailure {
+                    expect(writeConcernFailure.message.lowercased()).to(contain(errorContains))
+                }
+            } else if let logicError = error as? LogicError {
+                expect(logicError.errorDescription.lowercased()).to(contain(errorContains))
+            } else if let invalidArgumentError = error as? InvalidArgumentError {
+                expect(invalidArgumentError.errorDescription.lowercased()).to(contain(errorContains))
+            } else if let connectionError = error as? ConnectionError {
+                expect(connectionError.message.lowercased()).to(contain(errorContains))
+            } else {
+                XCTFail("\(error) does not contain message")
+            }
+        }
+    }
+
+    // swiftlint:enable cyclomatic_complexity
+
+    internal func checkCodeName(_ error: Error) throws {
+        if let errorCodeName = self.errorCodeName {
+            if let commandError = error as? CommandError {
+                expect(commandError.codeName).to(satisfyAnyOf(equal(errorCodeName), equal("")))
```

Comment:
if a codeName was provided by the test file, I don't think we can accept the empty string as valid also.

Reply:
I spoke with @kmahar about this earlier and these empty strings are necessary until SERVER-36755 is resolved. I've added a TODO note in the `checkCodeName` function as a reminder.